src/templates/spline.js

```diff
@@ -0,0 +1,108 @@
+class CubicSpline {
```

Comment:
It does feel so strange to me that the PlayCanvas Engine itself doesn't provide comprehensive support for curves. Keep in the back of your mind a way for us to migrate this at some point to that repo, ideally.